xdsl/transforms/individual_rewrite.py

```diff
@@ -17,10 +17,11 @@ class AdditionOfSameVariablesToMultiplyByTwo(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: arith.AddiOp, rewriter: PatternRewriter) -> None:
         if op.lhs == op.rhs:
-            assert isinstance(op.lhs.type, IntegerType | IndexType)
+            assert isinstance(type := op.lhs.type, IntegerType | IndexType)
+            value = 0 if isinstance(type, IntegerType) and type.bitwidth <= 1 else 2
             rewriter.replace_matched_op(
                 [
-                    li_op := arith.ConstantOp(IntegerAttr(2, op.lhs.type)),
+                    li_op := arith.ConstantOp(IntegerAttr(value, type)),
                     arith.MuliOp(op.lhs, li_op),
                 ]
             )
```

Comment:
isn't the whole thing just 0 if the bitwidth is 1? So could branch earlier and explicitly have it as a special case

Reply:
Could do